### ISSUE TITLE
@typeparam no longer allows trailing semi-colons

### ISSUE THREAD

**Issue by majorimi** [dotnet/aspnetcore#33602]
I have a open sourced [Blazor component library](https://github.com/majorimi/blazor-components). And recently I'm getting this error below when I try to run with `dotnet watch run` my demo app to develop the components.

```
C:\Program Files\dotnet\sdk\6.0.100-preview.4.21255.9\Microsoft.Common.CurrentVersion.targets(4964,5): 
error MSB3030: Could not copy the file "D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.CssEvents\obj\Debug\net5.0\staticwebassets\Majorsoft.Blazor.Components.CssEvents.StaticWebAssets.xml" 
because it was not found. [D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Modal\Majorsoft.Blazor.Components.Modal.csproj]
```

To give more details. I'm using Visual Studio 2019 (stable for development) but also have VS 2019 Preview installed. My projects using .NET 5.0.3 and `dotnet watch run` was working perfectly before .NET 6 preview got installed. But now `dotnet` commands using preview version: `dotnet\sdk\6.0.100-preview.4.21255.9` even if the project is just .NET 5.

As I can see the problem is `obj\Debug\net5.0\staticwebassets\<PROJECT_NAME>.StaticWebAssets.xml` is generated on every VS build. But when `watch` tool is detecting changes and does a "build" these files are deleted from `obj` folder. So instead of  file `copy` the watch tool might `move/delete` these files. I have multiple project in my solution this is true for all. And getting this error for lot of them...

**Comment by majorimi** [comment]
@mkArtakMSFT Now .NET 6 Preview 5 is out and tested `dotnet watch run` with Preview 5 and still has the error... This time with different projects. It seems issue happens randomly with a few projects in the solution.

```
C:\Program Files\dotnet\sdk\6.0.100-preview.5.21302.13\Microsoft.Common.CurrentVersion.targets(4964,5): error MSB3030: 
Could not copy the file "D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Timer\obj\Debug\net5.0\staticwebassets\Majorsoft.Blazor.Components.Timer.StaticWebAssets.xml" 
because it was not found. [D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Debounce\Majorsoft.Blazor.Components.Debounce.csproj]
```

**Also now Preview 5 has more issues with Templated components and Typeparams. I'm getting "build" errors with `watch` command. Meanwhile Visual Studio does not have build errors since it is compiling the project with .NET 5. But `watch run` uses .NET 6 Preview 5. See new errors below:**

```
D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Typeahead\TypeaheadInput.razor(1,17): 
error RZ1011: The 'typeparam' directives value(s) must be separated by whitespace. [D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Typeahead\Majorsoft.Blazor.Components.Typeahead.csproj]

D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Typeahead\TypeaheadInputText.razor(1,17): 
error RZ1011: The 'typeparam' directives value(s) must be separated by whitespace. [D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Typeahead\Majorsoft.Blazor.Components.Typeahead.csproj]
```

**Is this new issue should be posted separately? Also any quick solution to this?**
Thanks.

**Comment by majorimi** [comment]
@mkArtakMSFT  For unblocking my community work for now created a [global.json ](https://docs.microsoft.com/en-us/dotnet/core/tools/global-json?tabs=netcore3x#examples)file in the demo app root folder. Where set `"allowPrerelease": false` so now `dotnet watch run` uses .NET 5.

**But for sure in .NET 6 there are some issues with `dotnet watch run`. Also it seems that there will be breaking changes. Or `error RZ1011: The 'typeparam' directives value(s) must be separated by whitespace.` is also a bug in .NET 6?**

**Comment by pranavkm** [comment]
@majorimi the SDK will roll-forward to using the latest version installed, it's not influenced by the version of the app. However, they're designed to be back-compatible. I tried running `demo\Majorsoft.Blazor.Components.DemoApp` using the latest nightly of the .NET 6.0 preview SDK (https://github.com/dotnet/installer#installers-and-binaries) and VS 16.11.0 Preview 2.0 I'm not seeing the warning / errors you listed. 

Could you share detailed steps to reproduce the issue?

**Comment by majorimi** [comment]
Hi @pranavkm,

Thanks for your reply. Yes the Demo project will work since it does not have direct dependencies to any "Razor Class Library" projects in this solution. `demo\Majorsoft.Blazor.Components.DemoApp` uses already published Nuget packages from [Nuget.org](https://www.nuget.org/profiles/Blazor.Components). Demo project works for me as well. Please check `TestApps` solution folder. You can find 3 projects there. **Please try to run `Majorsoft.Blazor.Components.TestApp` you can even find `run.bat` inside the project folder.**

**For more details please check the project structure:**
- `demo\Majorsoft.Blazor.Components.DemoApp`: Published WASM application for demo purposes and uses already published Nuget packages as it would be used by a "package consumer" projects. This project is making sure all published Nuget packages are working as expected. **No direct references to any other projects in this SLN and It works!**
- `src\Majorsoft.Blazor.Components.TestApps.Common`:  this project **has references for all Component and Extension in the solution** and provides usage examples and other components with `\page` directive for navigation. This is **shared between WASM and Server hosted Blazor app**. With this project sample code is not duplicated.
- `src\Majorsoft.Blazor.Components.TestApp`: **Blazor WASM app using the `Common` project**. So this has direct references (via common project) to all components and during component development I can make sure all components works with Blazor WASM. **This does not work!! Please try `dotnet watch` with this project.**
- `src\Majorsoft.Blazor.Components.TestServerApp`: **Blazor Server Hosted application using the `Common` project** and during component development I can make sure all components works with Server side Blazor.

**Comment by pranavkm** [comment]
Thanks for the instructions @majorimi. This looks like a change as a result of a recent change to the Razor compiler to add support for generic type constraints. Removing the trailing semi-colon in https://github.com/majorimi/blazor-components/blob/master/src/Majorsoft.Blazor.Components.Typeahead/TypeaheadInput.razor#L1 and https://github.com/majorimi/blazor-components/blob/master/src/Majorsoft.Blazor.Components.Typeahead/TypeaheadInputText.razor#L1 fixes this issue. The fact that most directives supported trailing semi-colon is incidental, so I'll leave it up to our triage to determine if this is something we want to fix.

Once I fix this, I'm able to use watch just fine.

**Comment by majorimi** [comment]
Hi @pranavkm,
Thanks for the fix. **I was working for the second issue. Which was introduced by `Preview 5` Type param error**. However the original problem with XML comment files is still a problem:

```
C:\Program Files\dotnet\sdk\6.0.100-preview.5.21302.13\Microsoft.Common.CurrentVersion.targets(4964,5): error MSB3030: Could not copy the file "D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.CssEvents\obj\Debug\net5.0\staticwebassets\Majorsoft.Blazor.Components.CssEvents.StaticWebAssets.xml" because it was not found. [D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Modal\Majorsoft.Blazor.Components.Modal.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.5.21302.13\Microsoft.Common.CurrentVersion.targets(4964,5): error MSB3030: Could not copy the file "D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Core\obj\Debug\net5.0\staticwebassets\Majorsoft.Blazor.Components.Core.StaticWebAssets.xml" because it was not found. [D:\Projects\GitHub\blazor-components\src\Majorsoft.Blazor.Components.Modal\Majorsoft.Blazor.Components.Modal.csproj]
```

**Did you not encounter these errors?? What can be the reason for this?** Also these are appearing "randomly" and each time:
- I run Build from Visual Studio `Majorsoft.Blazor.Components.<projName>.StaticWebAssets.xml` is generated
- Save a file in VS and `dotnet watch run` kicks in and do a build but removed that XML, 1 error gone
- Run a build from VS -> file created
- CTRL + S in VS and dotnet watch kicks in and 1 error is gone...

I think you get the pattern :) and if I run this cycle enough time `dotnet watch run` will work fine.... Did you done something similar?

**Comment by mayur-ekbote** [comment]
Can you please add it to the breaking changes documentation?
